### ISSUE TITLE
Infinite loading of Specify when deleting auditing data

### ISSUE THREAD

**Issue by maxxxxxdlp** [specify/specify6#651]
I tested how long it takes to delete an auditing data when logging in into the database.

I created a new Botany database, made some edits to get 7 records in `auditlogfields` and 40 in `auditlog`. Then exported the database and duplicated these records to get **500 000 records in each** of those two tables (total **1.5 GB of auditng data**). Each record had unique ids. Loggin in into that database worked and querying the auditing database worked fine and I did not notice a performance hit.

The year for all of the records was set to 2010. Then I set the duration to save auditing data to 1 month and tried to log in into the database again. After 5 minutes of waiting, I forced Specify to quit and restarted MySQL server. Then launched them again. Waiting for 8 minutes did not lead to me getting into specify

When logging in, we can see normal window for normal duration of time:
<img width="437" alt="Screen Shot 2020-03-05 at 5 59 25 PM" src="https://user-images.githubusercontent.com/40512816/76114413-9f9af380-5fab-11ea-8cf6-945cdad02800.png">

But then it disappears and Specify becomes launched only in the background.
Screenshot from the activity monitor:
<img width="1440" alt="Screen Shot 2020-03-05 at 6 04 29 PM" src="https://user-images.githubusercontent.com/40512816/76113902-99f0de00-5faa-11ea-955e-ef01b92a7920.png">

It should be noted that I received foreign keys check errors when inserting data, so I run this command before insertion:
```
SET FOREIGN_KEY_CHECKS=0;
```
I hope it was not the cause of the infinite loading

These are the two queries I run repetitly to produce 500 000 records:
```
INSERT INTO `spauditlogfield`(`TimestampCreated`,`TimestampModified`,`Version`,`FieldName`,`NewValue`,`OldValue`,`ModifiedByAgentID`,`SpAuditLogID`,`CreatedByAgentID`) VALUES ('2010-03-04 17:27:29','2010-03-04 17:27:29',0,'lat1text','2° N',NULL,1,40,1),('2010-03-04 17:27:29','2010-03-04 17:27:29',0,'latLongType','Point',NULL,1,40,1),('2010-03-04 17:27:29','2010-03-04 17:27:29',0,'latitude1','2',NULL,1,40,1),('2010-03-04 17:27:29','2010-03-04 17:27:29',0,'localityName','2','1',1,40,1),('2010-03-04 17:27:29','2010-03-04 17:27:29',0,'long1text','2° W',NULL,1,40,1),('2010-03-04 17:27:29','2010-03-04 17:27:29',0,'longitude1','-2.0',NULL,1,40,1),('2010-03-04 17:27:29','2010-03-04 17:27:29',0,'originalLatLongUnit','0',NULL,1,40,1);
```
```
INSERT INTO `spauditlog`(`TimestampCreated`,`TimestampModified`,`Version`,`Action`,`ParentRecordId`,`ParentTableNum`,`RecordId`,`RecordVersion`,`TableNum`,`CreatedByAgentID`,`ModifiedByAgentID`) VALUES ('2010-03-04 17:25:15','2010-03-04 17:25:15',0,1,NULL,NULL,1,2,94,1,1),('2010-03-04 17:25:15','2010-03-04 17:25:15',0,1,1,NULL,1,1,8,1,1),('2010-03-04 17:25:15','2010-03-04 17:25:15',0,1,1,94,1,2,59,1,1),('2010-03-04 17:25:15','2010-03-04 17:25:15',0,1,NULL,NULL,1,1,60,1,1),('2010-03-04 17:25:15','2010-03-04 17:25:15',0,1,1,59,2,1,60,1,1),('2010-03-04 17:25:15','2010-03-04 17:25:15',0,1,2,59,3,1,60,1,1),('2010-03-04 17:25:15','2010-03-04 17:25:15',0,1,3,59,4,1,60,1,1),('2010-03-04 17:25:15','2010-03-04 17:25:15',0,1,NULL,NULL,1,3,94,1,1),('2010-03-04 17:25:15','2010-03-04 17:25:15',0,1,1,NULL,1,2,8,1,1),('2010-03-04 17:25:15','2010-03-04 17:25:15',0,1,1,94,1,3,59,1,1),('2010-03-04 17:25:15','2010-03-04 17:25:15',0,1,3,59,4,2,60,1,1),('2010-03-04 17:25:15','2010-03-04 17:25:15',0,1,2,59,3,2,60,1,1),('2010-03-04 17:25:15','2010-03-04 17:25:15',0,1,NULL,NULL,1,2,60,1,1),('2010-03-04 17:25:15','2010-03-04 17:25:15',0,1,1,59,2,2,60,1,1),('2010-03-04 17:25:16','2010-03-04 17:25:16',0,1,NULL,NULL,1,4,94,1,1),('2010-03-04 17:25:16','2010-03-04 17:25:16',0,1,1,NULL,1,3,8,1,1),('2010-03-04 17:25:16','2010-03-04 17:25:16',0,1,1,94,1,4,59,1,1),('2010-03-04 17:25:16','2010-03-04 17:25:16',0,1,NULL,NULL,1,3,60,1,1),('2010-03-04 17:25:16','2010-03-04 17:25:16',0,1,1,59,2,3,60,1,1),('2010-03-04 17:25:16','2010-03-04 17:25:16',0,1,2,59,3,3,60,1,1),('2010-03-04 17:25:16','2010-03-04 17:25:16',0,1,3,59,4,3,60,1,1),('2010-03-04 17:25:17','2010-03-04 17:25:17',0,1,1,94,2,3,96,1,1),('2010-03-04 17:25:17','2010-03-04 17:25:17',0,1,2,96,3,2,26,1,1),('2010-03-04 17:25:17','2010-03-04 17:25:17',0,1,3,26,1,2,44,1,1),('2010-03-04 17:25:17','2010-03-04 17:25:17',0,1,3,26,1,1,47,1,1),('2010-03-04 17:25:17','2010-03-04 17:25:17',0,1,3,26,1,1,47,1,1),('2010-03-04 17:25:17','2010-03-04 17:25:17',0,1,NULL,NULL,1,2,76,1,1),('2010-03-04 17:25:17','2010-03-04 17:25:17',0,1,2,76,3,1,77,1,1),('2010-03-04 17:25:17','2010-03-04 17:25:17',0,1,3,76,4,1,77,1,1),('2010-03-04 17:25:17','2010-03-04 17:25:17',0,1,4,76,5,1,77,1,1),('2010-03-04 17:25:17','2010-03-04 17:25:17',0,1,5,76,6,1,77,1,1),('2010-03-04 17:25:17','2010-03-04 17:25:17',0,1,6,76,7,1,77,1,1),('2010-03-04 17:25:17','2010-03-04 17:25:17',0,1,7,76,8,1,77,1,1),('2010-03-04 17:25:17','2010-03-04 17:25:17',0,1,8,76,9,1,77,1,1),('2010-03-04 17:25:17','2010-03-04 17:25:17',0,1,9,76,10,1,77,1,1),('2010-03-04 17:25:17','2010-03-04 17:25:17',0,1,NULL,NULL,1,1,77,1,1),('2010-03-04 17:25:17','2010-03-04 17:25:17',0,1,1,76,2,1,77,1,1),('2010-03-04 17:25:17','2010-03-04 17:25:17',0,1,3,26,4,3,23,1,1),('2010-03-04 17:27:23','2010-03-04 17:27:23',0,0,3,26,1,1,2,1,1),('2010-03-04 17:27:29','2010-03-04 17:27:29',0,1,3,26,1,2,2,1,1);
```

**Comment by timo11** [comment]
Before the fix, it took like 10 minutes on my machine to delete 5 million spauditlog recs from the ku ento db. 
Now the delete is truly a background task, and Specify starts normally, but it is much more than 10 minutes before the delete is executed. 
It would probably be good to test if data entry and logging work OK while the delete task is running. It's possible locks on the spauditlog table will cause problems.
There is a dump of the kuento auditlog in Databases/FilesForProgrammers/audlog.sql. It can be loaded into any db with: mysql -uROOTUSER -pROOTPW DBNAME < audlog.sql.
Having messed up foreign keys is probably OK for the auditlog tables, but might be a problem. The ento log is bigger and better except that there are no spauditlogfield entries.

**Comment by maxxxxxdlp** [comment]
TL;DR: works fine

Tested this on Friday with empty. botany database with `audlog.sql` file imported

After settings the month to save data for to 1 and restarting Specify, I logged in properly

The myslqd process was running in the background and making Specify be very slow
For several minutes the number of records in that table did not decrease even though the computer was busy

<img width="508" alt="Screen Shot 2020-03-06 at 3 10 26 PM" src="https://user-images.githubusercontent.com/40512816/76214189-3816bb00-61da-11ea-9625-9425350eea1f.png">

After creating a few localities, the count increased:
<img width="508" alt="Screen Shot 2020-03-06 at 3 12 52 PM" src="https://user-images.githubusercontent.com/40512816/76214196-3b11ab80-61da-11ea-9265-1c1cb68c813e.png">

After ~8min the myslqd process went back to using a small percentage of CPU and Specify started running much faster. The auditing table had new records for Localities added and all of the older rows removed
